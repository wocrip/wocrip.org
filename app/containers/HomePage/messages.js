import { defineMessages } from 'react-intl'


const scope = 'HomePage'

export default defineMessages({
  metaTitle: {
    id: `${scope}.metaTitle`,
    defaultMessage: 'Home',
  },
  metaDescription: {
    id: `${scope}.metaDescription`,
    defaultMessage: 'Modular infrastructure to build applications. For web, iOs, Android and desktop with a complete set of tools from design principles to production.',
  },
  headerTitle: {
    id: `${scope}.headerTitle`,
    defaultMessage: 'Build modular applications',
  },
  headerSubTitle: {
    id: `${scope}.headerSubTitle`,
    defaultMessage: `For web, mobile & desktop from design principles to production`,
  },
  Feature1Title: {
    id: `${scope}.Feature1Title`,
    defaultMessage: 'Business In Mind',
  },
  Feature1Text: {
    id: `${scope}.Feature1Text`,
    defaultMessage: 'Build applications for multiple platforms with ready to use features, servers and User Interface. It increase significantly the productivity.',
  },
  Feature2Title: {
    id: `${scope}.Feature2Title`,
    defaultMessage: 'Cutting Edge Technology',
  },
  Feature2Text: {
    id: `${scope}.Feature2Text`,
    defaultMessage: 'Offline-first React apps with CSS in JS for the front-end backed by AI on a distributed and shared platform of services, servers and microservices.',
  },
  Feature3Title: {
    id: `${scope}.Feature3Title`,
    defaultMessage: 'Optimized User Experience',
  },
  Feature3Text: {
    id: `${scope}.Feature3Text`,
    defaultMessage: 'With best practices applied in ergonomics, interactions, usability and accessibility. Available out of the box and customizable.',
  },
  onInviteOnly: {
    id: `${scope}.onInviteOnly`,
    defaultMessage: 'On invite only',
  },
  onInviteOnlySub: {
    id: `${scope}.onInviteOnlySub`,
    defaultMessage: `By subscribing you might get an invite. ${'\n'}And you will know when we open source it.`,
  },
  receiveNews: {
    id: `${scope}.receiveNews`,
    defaultMessage: 'I would like to receive news about Ceacle Modo.',
  },
})
