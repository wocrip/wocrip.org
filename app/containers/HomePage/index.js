import React from 'react'
import { Helmet } from 'react-helmet'
import { Flex, Box } from '@rebass/grid'
import { injectIntl } from 'react-intl'
import PropTypes from 'prop-types'

import Footer from 'components-ceacle/modules/Footer'

import EmailSubscriptionForm from 'components-shared/modules-form/EmailSubscriptionForm'
import Title from 'components-shared/basics/Title'
import Feature from 'components-shared/modules/Feature'
import HeaderHome from 'components-shared/templates/HeaderHome'
import Block from 'components-shared/basics/Block'
import Container from 'components-shared/basics/Container'

import messages from './messages'


const HomePage = ({
  intl: { formatMessage },
}) => (
  <Block>
    <Helmet>
      <title>{formatMessage(messages.metaTitle)}</title>
      <meta name="description" content={formatMessage(messages.metaDescription)} />
    </Helmet>

    <Container px={[30, 50, 80]}>
      <HeaderHome
        title={formatMessage(messages.headerTitle)}
        sub={formatMessage(messages.headerSubTitle)}
        width={[1]}
        textSize="h2"
        padding="50px 0"
        align="center"
        subBold
        subNoColor
        subMultiLine
      />
    </Container>

    <Container px={[30, 50, 80]}>
      <Flex
        flexWrap="wrap"
        px={[30, 50, 80]}
        pt={20}
        justifyContent="space-evenly"
      >
        <Box
          width={[1, 0.6, 0.5, 0.33]}
          px={[10, 10, 25]}
          mb={[50, 50, 20, 10]}
        >
          <Feature
            title={formatMessage(messages.Feature1Title)}
            text={formatMessage(messages.Feature1Text)}
            textSize={1.1}
            image="https://cdn1.ceacle.com/modo/images/illustrations/graphs@2x.png"
            imageWidth="100px"
            align="center"
          />
        </Box>
        <Box
          width={[1, 0.6, 0.5, 0.33]}
          px={[10, 10, 25]}
          mb={[50, 50, 20, 10]}
        >
          <Feature
            title={formatMessage(messages.Feature2Title)}
            text={formatMessage(messages.Feature2Text)}
            textSize={1.1}
            image="https://cdn1.ceacle.com/modo/images/illustrations/goup@2x.png"
            imageWidth="100px"
            align="center"
          />
        </Box>
        <Box
          width={[1, 0.6, 0.5, 0.33]}
          px={[10, 10, 25]}
          mb={[50, 50, 20, 10]}
        >
          <Feature
            title={formatMessage(messages.Feature3Title)}
            text={formatMessage(messages.Feature3Text)}
            textSize={1.1}
            image="https://cdn1.ceacle.com/modo/images/illustrations/users@2x.png"
            imageWidth="100px"
            align="center"
          />
        </Box>
      </Flex>
    </Container>

    <Container
      px={[30, 50, 80]}
      py={80}
    >
      <Flex
        flexWrap="wrap"
      >
        <Box
          width={[1, 0.8, 0.8, 0.5]}
          m="auto"
        >
          <Title
            sub={formatMessage(messages.onInviteOnlySub)}
            subColor="gray"
            subMultiLine
            align="center"
          >
            {formatMessage(messages.onInviteOnly)}
          </Title>

          <EmailSubscriptionForm
            application="modo"
            consentTexts={[
              {
                text: formatMessage(messages.receiveNews),
                key: 'receiveNews',
                optional: false,
              },
            ]}
            list="Modo"
          />
        </Box>
      </Flex>
    </Container>

    <Footer />
  </Block>
  )

HomePage.propTypes = {
  intl: PropTypes.object,
}

export default injectIntl(HomePage)
