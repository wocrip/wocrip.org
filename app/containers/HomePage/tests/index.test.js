import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import HomePage from '../index'


const renderComponent = (props = {}) => shallow(
  <HomePage {...props} />
)

describe('<HomePage />', () => {
  it('renders a <article /> tag', () => {
    const renderedComponent = renderComponent()
    expect(renderedComponent.type()).toEqual('article')
  })

  it('should render when reversed is true', () => {
    const renderedComponent = renderComponent()
    expect(renderedComponent.type()).toEqual('article')
  })

  it('should render', () => {
    const tree = renderer.create(
      <HomePage />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
