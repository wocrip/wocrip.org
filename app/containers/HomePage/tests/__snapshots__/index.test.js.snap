// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render 1`] = `
.c0 {
  box-sizing: border-box;
  width: 100%;
  padding-left: 30px;
  padding-right: 30px;
}

.c2 {
  box-sizing: border-box;
  width: 0%;
}

.c4 {
  box-sizing: border-box;
  width: 100%;
  padding-right: 20px;
}

.c3 {
  height: 100%;
  max-height: 700px;
  overflow: hidden;
}

.c3 svg {
  height: 100%;
  width: 150%;
}

.c3 path,
.c3 polygon {
  stroke-width: 4px;
  stroke-linecap: round;
  stroke-linejoin: round;
}

.c3 g:nth-child(1) {
  stroke: #0000b2;
  fill: #000019;
}

.c3 g:nth-child(2) {
  stroke: blue;
  fill: #000019;
}

.c3 g:nth-child(3) {
  stroke: #b3b3b3;
  fill: #262626;
}

.c1 {
  font-size: 3.5em;
  line-height: 1.2em;
  padding: 100px 0;
}

.c1 span {
  font-size: 0.6em;
  display: block;
  line-height: 1.5em;
  font-weight: 400;
  padding-top: 15px;
  color: #808080;
}

@media screen and (min-width:40em) {
  .c0 {
    width: 100%;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    width: 100%;
  }
}

@media screen and (min-width:64em) {
  .c0 {
    width: 66.66666666666666%;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    padding-left: 50px;
    padding-right: 50px;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    padding-left: 80px;
    padding-right: 80px;
  }
}

@media screen and (min-width:40em) {
  .c2 {
    width: 0%;
  }
}

@media screen and (min-width:52em) {
  .c2 {
    width: 0%;
  }
}

@media screen and (min-width:64em) {
  .c2 {
    width: 33.33333333333333%;
  }
}

@media screen and (min-width:40em) {
  .c4 {
    width: 100%;
  }
}

@media screen and (min-width:52em) {
  .c4 {
    width: 33.33333333333333%;
  }
}

@media screen and (min-width:40em) {
  .c4 {
    padding-right: 20px;
  }
}

@media screen and (min-width:52em) {
  .c4 {
    padding-right: 50px;
  }
}

@media (max-width:1020px) {
  .c3 {
    display: none;
  }
}

@media (max-width:1000px) {
  .c1 {
    font-size: 3.5em;
    padding: 70px 0;
  }
}

@media (max-width:900px) {
  .c1 {
    font-size: 2.5em;
    padding: 40px 0;
  }
}

<article>
  <span>
    <div
      className="c0"
      is={null}
    >
      <h2
        className="c1"
      >
        Modular infrastructure to build universal applications
        <span>
          For web, iOs, Android and desktop with a complete set of tools from design principles to production.
        </span>
      </h2>
    </div>
    <div
      className="c2"
      is={null}
    >
      <div
        className="c3"
      >
        <svg
          viewBox="0 0 402 372"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <polygon
              points="139,210.6 139,226.6 84,258.5 84,242.6"
            />
            <polygon
              points="84,242.6 84,258.5 28.6,226.6 28.6,210.6"
            />
            <polygon
              points="139,210.6 84,242.6 28.6,210.6 83.7,178.7"
            />
            <path
              d="M103.3,201l0,9.7c0,2.9-1.9,5.7-5.7,7.9c-7.6,4.4-20,4.4-27.6,0c-3.8-2.2-5.8-5.1-5.8-8l0-9.7 c0,2.9,1.9,5.8,5.8,8c7.6,4.4,20,4.4,27.6,0C101.4,206.7,103.3,203.8,103.3,201"
            />
            <path
              d="M97.6,192.8c7.6,4.4,7.7,11.6,0.1,16c-7.6,4.4-20,4.4-27.6,0c-7.6-4.4-7.7-11.6-0.1-16 C77.6,188.4,89.9,188.4,97.6,192.8z"
            />
          </g>
          <g>
            <polygon
              points="415.6,178.4 360.6,210.4 305.2,178.4 360.2,146.4"
            />
            <path
              d="M379.9,168.7l0,9.7c0,2.9-1.9,5.7-5.7,7.9c-7.6,4.4-20,4.4-27.6,0c-3.8-2.2-5.8-5.1-5.8-8l0-9.7 c0,2.9,1.9,5.8,5.8,8c7.6,4.4,20,4.4,27.6,0C378,174.5,379.9,171.6,379.9,168.7z"
            />
            <path
              d="M374.2,160.6c7.6,4.4,7.7,11.6,0.1,16c-7.6,4.4-20,4.4-27.6,0c-7.6-4.4-7.7-11.6-0.1-16 C354.2,156.2,366.5,156.2,374.2,160.6z"
            />
            <polygon
              points="471,210.4 416,242.4 360.6,210.4 415.7,178.4"
            />
            <path
              d="M435.3,200.7l0,9.7c0,2.9-1.9,5.7-5.7,7.9c-7.6,4.4-20,4.4-27.6,0c-3.8-2.2-5.8-5.1-5.8-8l0-9.7 c0,2.9,1.9,5.8,5.8,8c7.6,4.4,20,4.4,27.6,0C433.5,206.4,435.3,203.6,435.3,200.7z"
            />
            <path
              d="M429.6,192.6c7.6,4.4,7.7,11.6,0.1,16c-7.6,4.4-20,4.4-27.6,0c-7.6-4.4-7.7-11.6-0.1-16 C409.6,188.2,421.9,188.2,429.6,192.6z"
            />
            <polygon
              points="360.4,210.4 305.4,242.4 250,210.4 305,178.4"
            />
            <path
              d="M324.7,200.7l0,9.7c0,2.9-1.9,5.7-5.7,7.9c-7.6,4.4-20,4.4-27.6,0c-3.8-2.2-5.8-5.1-5.8-8l0-9.7 c0,2.9,1.9,5.8,5.8,8c7.6,4.4,20,4.4,27.6,0C322.8,206.4,324.7,203.6,324.7,200.7z"
            />
            <path
              d="M318.9,192.6c7.6,4.4,7.7,11.6,0.1,16c-7.6,4.4-20,4.4-27.6,0c-7.6-4.4-7.7-11.6-0.1-16 C298.9,188.2,311.3,188.2,318.9,192.6z"
            />
            <polygon
              points="415.8,242.4 415.8,258.3 360.8,290.2 360.8,274.3"
            />
            <polygon
              points="415.8,242.4 360.8,274.3 305.4,242.4 360.4,210.4"
            />
            <path
              d="M380.1,232.7l0,9.7c0,2.9-1.9,5.7-5.7,7.9c-7.6,4.4-20,4.4-27.6,0c-3.8-2.2-5.8-5.1-5.8-8l0-9.7 c0,2.9,1.9,5.8,5.8,8c7.6,4.4,20,4.4,27.6,0C378.2,238.4,380.1,235.5,380.1,232.7z"
            />
            <path
              d="M374.4,224.5c7.6,4.4,7.7,11.6,0.1,16c-7.6,4.4-20,4.4-27.6,0c-7.6-4.4-7.7-11.6-0.1-16 C354.4,220.1,366.7,220.1,374.4,224.5z"
            />
            <polygon
              points="305.4,242.4 250.4,274.3 195,242.4 250,210.4"
            />
            <path
              d="M269.7,232.7l0,9.7c0,2.9-1.9,5.7-5.7,7.9c-7.6,4.4-20,4.4-27.6,0c-3.8-2.2-5.8-5.1-5.8-8l0-9.7 c0,2.9,1.9,5.8,5.8,8c7.6,4.4,20,4.4,27.6,0C267.8,238.4,269.7,235.5,269.7,232.7z"
            />
            <path
              d="M263.9,224.5c7.6,4.4,7.7,11.6,0.1,16c-7.6,4.4-20,4.4-27.6,0c-7.6-4.4-7.7-11.6-0.1-16 C243.9,220.1,256.3,220.1,263.9,224.5z"
            />
            <polygon
              points="360.8,274.3 360.8,290.2 305.7,322.2 305.8,306.3"
            />
            <polygon
              points="360.8,274.3 305.8,306.3 250.4,274.3 305.4,242.4"
            />
            <path
              d="M325.1,264.6l0,9.7c0,2.9-1.9,5.7-5.7,7.9c-7.6,4.4-20,4.4-27.6,0c-3.8-2.2-5.8-5.1-5.8-8l0-9.7 c0,2.9,1.9,5.8,5.8,8c7.6,4.4,20,4.4,27.6,0C323.2,270.4,325.1,267.5,325.1,264.6z"
            />
            <path
              d="M319.3,256.5c7.6,4.4,7.7,11.6,0.1,16c-7.6,4.4-20,4.4-27.6,0c-7.6-4.4-7.7-11.6-0.1-16 C299.4,252.1,311.7,252.1,319.3,256.5z"
            />
            <polygon
              points="195.1,306.3 195.1,322.2 139.7,290.2 139.8,274.3"
            />
            <polygon
              points="250.2,274.3 195.1,306.3 139.8,274.3 194.8,242.4"
            />
            <path
              d="M214.5,264.6l0,9.7c0,2.9-1.9,5.7-5.7,7.9c-7.6,4.4-20,4.4-27.6,0c-3.8-2.2-5.8-5.1-5.8-8l0-9.7 c0,2.9,1.9,5.8,5.8,8c7.6,4.4,20,4.4,27.6,0C212.6,270.4,214.5,267.5,214.5,264.6"
            />
            <path
              d="M208.7,256.5c7.6,4.4,7.7,11.6,0.1,16c-7.6,4.4-20,4.4-27.6,0c-7.6-4.4-7.7-11.6-0.1-16 C188.7,252.1,201.1,252.1,208.7,256.5z"
            />
            <polygon
              points="305.6,306.3 305.5,322.2 250.5,354.2 250.6,338.3"
            />
            <polygon
              points="250.6,338.3 250.5,354.2 195.1,322.2 195.2,306.3"
            />
            <polygon
              points="305.6,306.3 250.6,338.3 195.2,306.3 250.2,274.3"
            />
            <polygon
              points="470.8,210.4 470.8,226.3 415.8,258.3 415.8,242.4"
            />
            <path
              d="M269.9,296.6l0,9.7c0,2.9-1.9,5.7-5.7,7.9c-7.6,4.4-20,4.4-27.6,0c-3.8-2.2-5.8-5.1-5.8-8l0-9.7 c0,2.9,1.9,5.8,5.8,8c7.6,4.4,20,4.4,27.6,0C268,302.3,269.9,299.5,269.9,296.6z"
            />
            <path
              d="M264.1,288.5c7.6,4.4,7.7,11.6,0.1,16c-7.6,4.4-20,4.4-27.6,0c-7.6-4.4-7.7-11.6-0.1-16 C244.1,284.1,256.5,284.1,264.1,288.5z"
            />
          </g>
          <g>
            <polygon
              points="359.5,83.2 359.4,99.1 304.4,131 304.4,115.1"
            />
            <polygon
              points="304.4,115.1 304.4,131 249.4,163 249.4,147.1"
            />
            <polygon
              points="249.2,147.1 249.2,163 194.2,195 194.2,179.1"
            />
            <polygon
              points="194.2,179.1 194.2,195 138.8,163 138.8,147.1"
            />
            <polygon
              points="359.5,67.2 359.4,83.1 304.4,115.1 304.4,99.1"
            />
            <polygon
              points="304.4,99.1 304.4,115.1 249.4,147 249.4,131.1"
            />
            <polygon
              points="249.2,131.1 249.2,147 194.2,179 194.2,163.1"
            />
            <polygon
              points="194.2,163.1 194.2,179 138.8,147 138.8,131.1"
            />
            <polygon
              points="359.5,51.2 359.4,67.1 304.4,99.1 304.4,83.2"
            />
            <polygon
              points="359.5,51.2 304.4,83.2 249.1,51.2 304.1,19.2"
            />
            <path
              d="M323.8,41.5l0,9.7c0,2.9-1.9,5.7-5.7,7.9c-7.6,4.4-20,4.4-27.6,0c-3.8-2.2-5.8-5.1-5.8-8l0-9.7 c0,2.9,1.9,5.8,5.8,8c7.6,4.4,20,4.4,27.6,0C321.9,47.2,323.8,44.4,323.8,41.5z"
            />
            <path
              d="M318,33.4c7.6,4.4,7.7,11.6,0.1,16c-7.6,4.4-20,4.4-27.6,0c-7.6-4.4-7.7-11.6-0.1-16 C298,29,310.4,29,318,33.4z"
            />
            <polygon
              points="304.4,83.2 304.4,99.1 249.4,131 249.4,115.1"
            />
            <polygon
              points="304.4,83.2 249.4,115.1 194.1,83.2 249.1,51.2"
            />
            <path
              d="M268.8,73.5l0,9.7c0,2.9-1.9,5.7-5.7,7.9c-7.6,4.4-20,4.4-27.6,0c-3.8-2.2-5.8-5.1-5.8-8l0-9.7 c0,2.9,1.9,5.8,5.8,8c7.6,4.4,20,4.4,27.6,0C266.9,79.2,268.7,76.3,268.8,73.5z"
            />
            <path
              d="M263,65.3c7.6,4.4,7.7,11.6,0.1,16c-7.6,4.4-20,4.4-27.6,0c-7.6-4.4-7.7-11.6-0.1-16 C243,60.9,255.3,60.9,263,65.3z"
            />
            <polygon
              points="249.2,115.1 249.2,131 194.2,163 194.2,147.1"
            />
            <polygon
              points="194.2,147.1 194.2,163 138.8,131 138.8,115.1"
            />
            <polygon
              points="249.2,115.1 194.2,147.1 138.8,115.1 193.8,83.2"
            />
            <path
              d="M213.5,105.4l0,9.7c0,2.9-1.9,5.7-5.7,7.9c-7.6,4.4-20,4.4-27.6,0c-3.8-2.2-5.8-5.1-5.8-8l0-9.7 c0,2.9,1.9,5.8,5.8,8c7.6,4.4,20,4.4,27.6,0C211.6,111.2,213.5,108.3,213.5,105.4z"
            />
            <path
              d="M207.8,97.3c7.6,4.4,7.7,11.6,0.1,16c-7.6,4.4-20,4.4-27.6,0c-7.6-4.4-7.7-11.6-0.1-16 C187.8,92.9,200.1,92.9,207.8,97.3z"
            />
          </g>
        </svg>
      </div>
    </div>
  </span>
  <span>
    <div
      className="c4"
      is={null}
    >
      <span />
    </div>
    <div
      className="c4"
      is={null}
    >
      <span />
    </div>
    <div
      className="c4"
      is={null}
    >
      <span />
    </div>
  </span>
</article>
`;
